Lib/dis.py

```diff
@@ -464,18 +467,27 @@ def print_instruction(self, instr, mark_as_current=False):
     def print_instruction_line(self, instr, mark_as_current):
         """Format instruction details for inclusion in disassembly output."""
         lineno_width = self.lineno_width
+        positions_width = self.positions_width
         offset_width = self.offset_width
         label_width = self.label_width
 
-        new_source_line = (lineno_width > 0 and
+        new_source_line = ((lineno_width > 0 or positions_width > 0) and
                            instr.starts_line and
                            instr.offset > 0)
         if new_source_line:
             print(file=self.file)
 
         fields = []
         # Column: Source code line number
-        if lineno_width:
+        if positions_width:
+            # reporting positions instead of just line numbers
+            if instr_positions := instr.positions:
+                ps = tuple('?' if p is None else p for p in instr_positions)
+                positions_str = "%s:%s-%s:%s" % (ps[0], ps[2], ps[1], ps[3])
+                fields.append(f'{positions_str:{positions_width}}')
```

Comment:
You're kind of mixing styles here, using %-formatting on one line and f-strings in the next. Can you just do the whole thing with an f-string?

Reply:
I can definitely do it. I'll do:

```py
positions_str = f"{ps[0]}:{ps[2]}-{ps[1]}:{ps[3]}"
fields.append(f'{positions_str:{positions_width}}')
```

I'd like to keep the `positions_str` on a separate line just to avoid nested f-strings which are somewhat hard to read.